graphql_compiler/compiler/emit_sql.py

```diff
@@ -126,6 +134,101 @@ def _find_folded_fields(ir):
+
+    This method is needed because if XML PATH column is selected with an alias then each
+    entry in the resulting array is wrapped in XML tags bearing that alias.
+    e.g. `SELECT (SELECT ... FOR XML PATH ('')) AS blah ...` results in a string of the form
+    <blah>element1</blah><blah>element2</blah>. Since SQLAlchemy adds labels to all Selectables, we
+    must to suppress that labeling.
+    """
+    if isinstance(element.element, XMLPathBinaryExpression):
+        # this indicates that no label should be inserted for this element
+        kw.update(
+            within_columns_clause=False  # label always gets added if within_columns_clause is True
+        )
+    return compiler.visit_label(element, **kw)
+
+
+def get_xml_path_clause(output_column, where):
+    r"""Produce an MSSQL-style XML PATH-based aggregation subquery.
+
+    XML PATH clause aggregates the values of the output_column from the output vertex
+    properly encoded to ensure that we can reconstruct a list of the original values
+    regardless of what characters they contain, whether they are empty or null, and
+    regardless of their data type.
+
+    Empty arrays are represented as '', null values are represented as '~', array
+    entries are delimited using '|'.
+
+    All occurrences of '^', '~', and '|' in the original string values are
+    replaced with '^e' (escape), '^n' (null), and '^d' (delimiter), resp.
+
+    Undoing the encoding above, as well as the XML reference entity encoding performed
+    by the XML PATH statement, is deferred to post-processing when the list is retrieved
+    from the string representation produced by the subquery.
+
+    Post-processing must split on '|', convert '~' to None, and undo both the encoding above
+    and the XML reference entity encoding auto performed by XML PATH. In particular, undo
+    '^d' -> '|', '^e' -> '^', '^n' -> '~', '&amp;' -> '&', '&gt;' -> '>', '&lt;' -> '<',
+    '&#xHEX;' -> '\xHEX'. For example, any string containing the
+    "ack acknowledge ctrl-f character", represented by hexadecimal 0x6, will be converted to
+    a string containing the HTML sequence "&#x6;" This conversion can be undone with chr(int("6", 16))
+    to convert the sequence above to the character '\x06'. Any sequence matching
+    &#x([A-Za-z0-9]+); can be converted to the proper unicode character representation likewise.
+    The remaining XML post processing can be done with
+    https://docs.python.org/3/library/html.html#html.unescape
+
+    Joining from the preceding vertex to the correct output vertex
+    is performed with the WHERE clause. This is contrasted with PostgreSQL subqueries,
+    in which the join from penultimate vertex to output vertex is done within
+    the FROM/JOIN clause of the aggregation subquery.
+
+    Args:
+        output_column: SQLAlchemy Column, value being aggregated with XML PATH
+        where: SQLAlchemy BinaryExpression, predicate used to filter
```

Comment:
How about calling this `predicate_expression`? That way it's both more expressive and also avoids the `.where(where)` bit of code on line 227.